packages/dashboards/tests/acceptance/dashboards-test.js

```diff
@@ -662,4 +662,140 @@ module('Acceptance | Dashboards', function(hooks) {
       'the filters reflect the dashboards modified filters'
     );
   });
+
+  test('New widget after clone', async function(assert) {
+    assert.expect(18);
+
+    let originalDashboardTitle, originalWidgetTitles;
+
+    await visit('/dashboards/1');
+
+    originalDashboardTitle = find('.page-title .clickable').textContent.trim();
+
+    originalWidgetTitles = findAll('.navi-widget__title').map(el => el.textContent.trim());
+
+    await click('.navi-icon__copy');
+
+    assert.equal(currentURL(), '/dashboards/6/view', 'Cloning a dashboard transitions to newly made dashboard');
+
+    assert
+      .dom('.page-title .clickable')
+      .hasText(
+        `Copy of ${originalDashboardTitle}`,
+        'Cloned dashboard has the same title as Original dashboard with `copy of` prefix title'
+      );
+
+    assert.deepEqual(
+      findAll('.navi-widget__title').map(el => el.textContent.trim()),
+      originalWidgetTitles,
+      'Cloned widgets are present in the dashboard '
+    );
+
+    // Check original set of widgets
+    const widgetsBefore = findAll('.navi-widget__title').map(el => el.textContent.trim());
+
+    assert.deepEqual(
+      widgetsBefore,
+      ['Mobile DAU Goal', 'Mobile DAU Graph', 'Mobile DAU Table'],
+      '"Untitled Widget" is not initially present on dashboard'
+    );
+
+    // Create new widget
+    await click('.add-widget .btn');
+    await click('.add-widget-modal .add-to-dashboard');
+
+    // Fill out request
+    await click($('.checkbox-selector--metric .grouped-list__item:contains(Total Clicks) .grouped-list__add-icon')[0]);
+
+    // Save without running
+    await click('.navi-report-widget__save-btn');
+    assert.ok(
+      currentURL().endsWith('/dashboards/6/view'),
+      'After saving without running, user is brought back to dashboard view'
+    );
+
+    let widgetsAfter = findAll('.navi-widget__title').map(el => el.textContent.trim());
+
+    assert.deepEqual(
+      widgetsAfter,
+      ['Mobile DAU Goal', 'Mobile DAU Graph', 'Mobile DAU Table', 'Untitled Widget'],
+      '"Untitled Widget" has been added to dashboard'
+    );
+
+    let widgetColumns = findAll('.navi-widget:nth-child(4) .table-header-cell').map(el => el.textContent.trim());
+
+    assert.deepEqual(
+      widgetColumns,
+      ['Date', 'Total Clicks'],
+      'Table columns for the new widget are rendered correctly'
+    );
+
+    // Create another new widget
+    await click('.add-widget .btn');
+    await click('.add-widget-modal .add-to-dashboard');
+
+    // Fill out request
+    await click(
+      $('.checkbox-selector--metric .grouped-list__item:contains(Total Page Views) .grouped-list__add-icon')[0]
+    );
+
+    // Run request
+    await click('.navi-report-widget__run-btn');
+    // Regex to check that a string ends with "{uuid}/view"
+    const TempIdRegex = /\/[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}\/view$/;
```

Comment:
Do we use this regex elsewhere? Maybe it should be in a util module and imported where it's needed.